src/ParseObject.js

```diff
@@ -2419,6 +2413,13 @@ const DefaultController = {
             const batch = [];
             const nextPending = [];
             pending.forEach(el => {
+              if (allowCustomObjectId && Object.prototype.hasOwnProperty.call(el, 'id') && !el.id) {
+                throw new ParseError(
+                  ParseError.MISSING_OBJECT_ID,
+                  'objectId must not be empty or null'
```

Comment:
This has changed from `'objectId must not be empty, null or undefined'` to ` 'objectId must not be empty or null'`. Why is `undefined` not allowed?